library/swift/Headers.swift

```diff
@@ -4,45 +4,58 @@ import Foundation
 /// To instantiate new instances, see `{Request|Response}HeadersBuilder`.
 @objcMembers
 public class Headers: NSObject {
```

Comment:
Can we merge the `Headers` type with the `HeadersContainer` type? I don't see any reason to add that additional layer, but maybe I'm missing a reason why it's necessary?

Reply:
One of my previous solutions was to merge `Headers` and `HeadersContainer` but I decided against it as:

- I wanted `HeadersContainers` to be a struct - not possible with `Headers` as it has to inherit from `NSObject`. There are some negative of using structs tho - more copying.
- `HeadersBuilder` depends on a lot of stuff that's currently in `HeadersContainer`. If we merge `Headers` and `HeadersContainer` then `HeadersBuilder `would either have to have duplicated logic between `Headers `and `HeadersBuilder` or make `HeadersBuilder` depend on `Headers`. I decided that having a separate `HeadersContainer` class was a better approach to ensure that all of the case-insensitive lookup logic lives in one place and is shared between `Headers` and `HeaderBuilder`.